back/engines/commercial/bulk_import_ideas/app/services/bulk_import_ideas/import_ideas_service.rb

```diff
@@ -0,0 +1,216 @@
+          next unless key.include? '_'
+
+          field, locale = key.split '_'
+          case field
+          when 'Title'
+            title_multiloc[locale] = value
+          when 'Body'
+            body_multiloc[locale] = value
+          end
+        end
+
+        idea_row[:title_multiloc]       = title_multiloc
+        idea_row[:body_multiloc]        = body_multiloc
+        idea_row[:topic_titles]         = (xlsx_row['Topics'] || '').split(';').map(&:strip).select { |topic| topic }
+        idea_row[:project_title]        = xlsx_row['Project']
+        idea_row[:user_email]           = xlsx_row['Email']
+        idea_row[:image_url]            = xlsx_row['Image URL']
+        idea_row[:phase_rank]           = xlsx_row['Phase']
+        idea_row[:published_at]         = xlsx_row['Date (dd-mm-yyyy)']
+        idea_row[:latitude]             = xlsx_row['Latitude']
+        idea_row[:longitude]            = xlsx_row['Longitude']
+        idea_row[:location_description] = xlsx_row['Location Description']
+        idea_row[:id]                   = xlsx_row['ID']
+        idea_row
+      end
+    end
+
+    private
+
+    def import_idea(idea_row)
+      idea_attributes = {}
+      add_title_multiloc idea_row, idea_attributes
+      add_body_multiloc idea_row, idea_attributes
+      add_project idea_row, idea_attributes
+      add_author idea_row, idea_attributes
+      add_published_at idea_row, idea_attributes
+      add_publication_status idea_row, idea_attributes
+      add_location idea_row, idea_attributes
+      add_phase idea_row, idea_attributes
+      add_topics idea_row, idea_attributes
+
+      idea = Idea.new idea_attributes
+      # Later iteration: parseable error using idea.errors.details
+      raise Error.new 'bulk_import_ideas_idea_not_valid', value: idea.errors.messages unless idea.valid?
+
+      idea.save!
+
+      create_idea_image idea_row, idea
+
+      idea
+    end
+
+    def add_title_multiloc(idea_row, idea_attributes)
+      raise Error.new 'bulk_import_ideas_blank_title', row: idea_row[:id] if idea_row[:title_multiloc].blank?
+
+      idea_attributes[:title_multiloc] = idea_row[:title_multiloc]
+    end
+
+    def add_body_multiloc(idea_row, idea_attributes)
+      raise Error.new 'bulk_import_ideas_blank_body', row: idea_row[:id] if idea_row[:body_multiloc].blank?
+
+      idea_attributes[:body_multiloc] = idea_row[:body_multiloc]
+    end
+
+    def add_project(idea_row, idea_attributes)
+      raise Error.new 'bulk_import_ideas_blank_project', row: idea_row[:id] if idea_row[:project_title].blank?
+
+      project_title = idea_row[:project_title].downcase.strip
+      # Later iteration: Only load necessary attributes? Preload all projects before importing all?
+      project = Project.all.find do |find_project|
+        find_project
+          .title_multiloc
+          .values
+          .map { |v| v.downcase.strip }
+          .include? project_title
+      end
+      if project.blank?
```

Comment:
`find` returns `nil` when nothing is found, so this can be simplified:
```suggestion
      if project
```